worker/caasadmission/k8s/admission.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package k8s
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/juju/errors"
+	admission "k8s.io/api/admissionregistration/v1beta1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/juju/juju/caas/kubernetes/provider"
+)
+
+type CertificateBroker interface {
+	// Register a new domain name lease on the brokers certificate. Returns a
+	// lease function if no errors
+	RegisterDomainLeases(...string) (func(), error)
+
+	// Set of root certificates in pem format
+	RootCertificates() ([]byte, error)
+}
+
+// TODO this is a tmp struct and won't make it to final
+type CertWatcherBroker struct {
+	CertWatcher func() *tls.Certificate
+}
+
+type K8sBroker interface {
+	EnsureMutatingWebhookConfiguration(*admission.MutatingWebhookConfiguration) (func(), error)
+}
+
+// Represents a creator of mutating webhooks that is context aware of the
+// current controller.
+type AdmissionCreator interface {
+	EnsureMutatingWebhookConfiguration() (func(), error)
+}
+
+// Func type of AdmissionCreator
+type AdmissionCreatorFunc func() (func(), error)
+
+const (
+	Component = "admission"
+)
+
+var (
+	anyMatch = []string{"*"}
+)
+
+// Implements AdmissionCreator interface for func type
+func (a AdmissionCreatorFunc) EnsureMutatingWebhookConfiguration() (func(), error) {
+	return a()
+}
+
+// Instantiates a new AdmissionCreator for the supplied context arguments.
+func NewAdmissionCreator(
+	certBroker CertificateBroker,
+	namespace, modelName, modelUUID string,
```

Comment:
`modelUUID` is not used here.
And no need to expose `namespace, modelName`  to worker package here.
Because broker has all of these, we can just ensure all the labels stuff inside `broker.EnsureMutatingWebhookConfiguration` under `k8s.go`.
The broker:caasadmissionWorker is 1:1 relationship.

Reply:
modelUUID has been removed, was left over after the labels chat.

The current broker code doesn't do this and I think it's good to be explicit of the requirements the controller wants. Current code focuses around app names more than anything.